request_test.go

```diff
@@ -317,10 +317,7 @@ func TestUnmarshal_nonNumericID(t *testing.T) {
 }
 
 func TestUnmarshalSetsAttrs(t *testing.T) {
-	out, err := unmarshalSamplePayload()
-	if err != nil {
-		t.Fatal(err)
-	}
+	out := unmarshalSamplePayload(t)
```

Comment:
I'm curious to know why in this test code, we want to start ignoring the unmarshaling error.

Reply:
I see. the `func unmarshalSamplePayload` declaration has changed ([L643](https://github.com/getoutreach/jsonapi/pull/2/files#diff-d7ac8930f8073cd3703fe4775ba3b5f3a312c3ecd8a2ab8f6c0da5bb6764da13R634)).

Then I'm curious, in what situation do we want to return error, in what situation do we want to throw a failure within a function? For this matter, is there a guideline/pattern to follow based on commonly agreed best practices in Go?

Reply:
I replied to you in slack as to why it's not being checked anymore! It's a paradigm specific to go testing